src/hatch_fancy_pypi_readme/_config.py

```diff
@@ -19,36 +22,97 @@ class Config:
     substitutions: list[Substituter]
 
 
+SCHEMA = {
+    "type": "object",
+    "properties": {
+        "content-type": {
+            "type": "string",
+            "enum": ["text/markdown", "text/x-rst"],
+        },
+        "fragments": {
+            "type": "array",
+            "minItems": 1,
+            # Items are validated separately for better error messages.
+            "items": {"type": "object"},
+        },
+        "substitutions": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "pattern": {"type": "string", "format": "regex"},
+                    "replacement": {"type": "string"},
+                    "ignore-case": {"type": "boolean"},
+                },
+                "required": ["pattern", "replacement"],
+                "additionalProperties": False,
+            },
+        },
+    },
+    "required": ["content-type", "fragments"],
+    "additionalProperties": False,
+}
+
+V = Draft202012Validator(
```

Comment:
(I didn't mean to literally suggest one usually calls this `V` :), `validator` or something is a usual name, in lowercase since it's an instance).